### PR TITLE
add delete pod

### PR DESCRIPTION
**Pull request by xlgao-zju**
A part of #8 

Signed-off-by: Xianglin Gao <xlgao@zju.edu.cn>

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/frakti/11)

### CODE REVIEW

File: pkg/hyper/hyper.go [hunk 1]
```diff
@@ -68,7 +70,16 @@ func (h *Runtime) StopPodSandbox(podSandBoxID string) error {
 // DeletePodSandbox deletes the sandbox. If there are any running containers in the
 // sandbox, they should be force deleted.
 func (h *Runtime) DeletePodSandbox(podSandBoxID string) error {
-	return fmt.Errorf("Not implemented")
+	request := types.PodRemoveRequest{
+		PodID: podSandBoxID,
+	}
+
+	_, err := h.client.client.PodRemove(context.Background(), &request)
```
Comment (feiskyer) [inline pkg/hyper/hyper.go]: @xlgao-zju Thanks for the PR. Need to process pod already nonexistent scenario here. If the sandbox has already been nonexistent, `DeletePodSandbox` should return nil.

**Comment by feiskyer** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[pkg/hyper/hyper.go, line 77 \[r2\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/frakti/11#-KP_7Lhmeu91UKqVGo_f:-KP_7Lhnh2j4VdvNHTtB:birxizz) ([raw file](https://github.com/kubernetes/frakti/blob/4b3027db76dd18189e23054bbf3535014cfb0049/pkg/hyper/hyper.go#L77)):*
> ```Go
> 	}
> 
> 	response, err := h.client.client.PodRemove(context.Background(), &request)
> ```

The rpc request should use a context with timeout, e.g.

```go
// getContextWithTimeout returns a context with timeout.
func getContextWithTimeout(timeout time.Duration) (context.Context, context.CancelFunc) {
	return context.WithTimeout(context.Background(), timeout)
}
```

---

*[pkg/hyper/hyper.go, line 78 \[r2\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/frakti/11#-KP_7bJqbcy4DH57jFSR:-KP_7bJqbcy4DH57jFSS:b-g7l589) ([raw file](https://github.com/kubernetes/frakti/blob/4b3027db76dd18189e23054bbf3535014cfb0049/pkg/hyper/hyper.go#L78)):*
> ```Go
> 
> 	response, err := h.client.client.PodRemove(context.Background(), &request)
> 	if response.Code == -2 {
> ```

Please avoid magic number here, define a const for `-2`.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/frakti/11)*

**Comment by xlgao-zju** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[pkg/hyper/hyper.go, line 77 \[r2\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/frakti/11#-KP_7Lhmeu91UKqVGo_f:-KPkdn1YmZVBA3_nY7Sk:b-896fix) ([raw file](https://github.com/kubernetes/frakti/blob/4b3027db76dd18189e23054bbf3535014cfb0049/pkg/hyper/hyper.go#L77)):*
<details><summary><i>Previously, feiskyer (Pengfei Ni) wrote…</i></summary>
> The rpc request should use a context with timeout, e.g.
> 
> ```go
> // getContextWithTimeout returns a context with timeout.
> func getContextWithTimeout(timeout time.Duration) (context.Context, context.CancelFunc) {
> 	return context.WithTimeout(context.Background(), timeout)
> }
> ```
> 
</details>
Done.

---

*[pkg/hyper/hyper.go, line 78 \[r2\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/frakti/11#-KP_7bJqbcy4DH57jFSR:-KPkdnjbJklHofsQ_Wb3:b-896fix) ([raw file](https://github.com/kubernetes/frakti/blob/4b3027db76dd18189e23054bbf3535014cfb0049/pkg/hyper/hyper.go#L78)):*
<details><summary><i>Previously, feiskyer (Pengfei Ni) wrote…</i></summary>
> Please avoid magic number here, define a const for `-2`.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/frakti/11)*

**Comment by feiskyer** [comment]
@xlgao-zju Thanks. LGTM.

**Pull request merged**